stake-pool/cli/src/main.rs

```diff
@@ -433,131 +501,124 @@ fn command_withdraw(
+
+            instructions.push(withdraw(
                 &spl_stake_pool::id(),
                 &pool,
                 &pool_withdraw_authority,
-                &stake_pubkey,
+                &withdraw_stake.pubkey,
+                &stake_receiver.unwrap(), // Cannot be none at this point
                 &config.owner.pubkey(),
                 &burn_from,
                 &pool_data.pool_mint,
                 &spl_token::id(),
                 &stake_program_id(),
+                withdraw_stake.amount,
             )?);
-            // Merge into stake_receiver (if present)
-            if let Some(merge_into) = stake_receiver {
-                instructions.push(merge_stake(
-                    &merge_into,
-                    &stake_pubkey,
-                    &config.owner.pubkey(),
-                ));
-            }
         } else {
-            // Withdraw to split account
-            let stake_receiver: Pubkey = match stake_receiver {
-                Some(value) => *value,
-                None => {
-                    // Account for tokens not specified, creating one
-                    println!(
-                        "Creating account to receive stake {}",
-                        stake_receiver_account.pubkey()
-                    );
-
-                    let stake_receiver_account_balance = config
-                        .rpc_client
-                        .get_minimum_balance_for_rent_exemption(STAKE_STATE_LEN)?;
-
-                    instructions.push(
-                        // Creating new account
-                        system_instruction::create_account(
-                            &config.fee_payer.pubkey(),
-                            &stake_receiver_account.pubkey(),
-                            stake_receiver_account_balance,
-                            STAKE_STATE_LEN as u64,
-                            &stake_program_id(),
-                        ),
-                    );
-
-                    signers.push(&stake_receiver_account);
-
-                    total_rent_free_balances += stake_receiver_account_balance;
-
-                    stake_receiver_account.pubkey()
-                }
-            };
-
-            instructions.push(withdraw(
+            // Claim to get whole account
+            instructions.push(claim(
                 &spl_stake_pool::id(),
                 &pool,
                 &pool_withdraw_authority,
-                &stake_pubkey,
-                &stake_receiver,
+                &withdraw_stake.pubkey,
                 &config.owner.pubkey(),
                 &burn_from,
                 &pool_data.pool_mint,
                 &spl_token::id(),
                 &stake_program_id(),
-                amount,
             )?);
-        }
 
-        let mut transaction =
-            Transaction::new_with_payer(&instructions, Some(&config.fee_payer.pubkey()));
+            // Merge into stake_receiver
+            if let Some(merge_into) = stake_receiver {
```

Comment:
If, for example, we don't provide `stake_receiver`, and our withdraw results in claiming two accounts and withdrawing from one, will this work as intended?  `stake_receiver` might not be defined at this point, but we might still want to merge into the final account.  Is there something in the sorting that makes this work out?  If so, we should clarify that somewhere

Reply:
Ok, this is a bit confusing.
If we have at lease one `withdraw` command then it will be first in the list (as per this comment) and it will definitely create `stake_receiver`:
`// This account will be withdrawn partially (if remaining_amount < account.lamports), so we put it first`
However if we do only `claim`s and do not specify `stake_receiver` then the stakes will not be joined.

So probably we need to rework params a bit. May be make this a required param? Or if we are in `claim`-only situation and have more than one account then merge all of them to the first one?

Reply:
I thought about it for a while and will do the second option (if there are only bunch of `claim`s then the first one of them is used to merge everything to).